src/Components/SceneList/Internal/SceneDialog.tsx

```diff
@@ -321,72 +356,83 @@ const SceneDialog: React.FC<ISceneDialogProps> = ({
     const handleFileChange = useCallback(
         (file: File) => {
             if (file) {
-                if (blobsInContainer.map((e) => e.Name).includes(file.name)) {
-                    setIsSelectedFileExistInBlob(true);
-                } else {
-                    setIsSelectedFileExistInBlob(false);
-                }
+                dispatch({
+                    type: SET_IS_SELECTED_FILE_EXIST_IN_BLOB,
+                    payload: state.blobsInContainer
+                        .map((e) => e.Name)
+                        .includes(file.name)
+                });
             } else {
-                setIsSelectedFileExistInBlob(false);
-                setIsOverwriteFile(false);
+                dispatch({
+                    type: RESET_OVERWRITE_FILE_AND_EXIST_IN_BLOB,
+                    payload: {
+                        isSelectedFileExistInBlob: false,
+                        isOverwriteFile: false
+                    }
+                });
             }
-            setSelectedFile(file);
+            dispatch({
+                type: SET_SELECTED_FILE,
+                payload: file
+            });
         },
-        [blobsInContainer]
+        [state.blobsInContainer]
     );
 
     const handleOnBlobsLoaded = (blobs: Array<IStorageBlob>) => {
-        setBlobsInContainer(blobs);
+        dispatch({
+            type: SET_BLOBS_IN_CONTAINER,
+            payload: blobs
+        });
     };
 
     const handleOnPivotClick = useCallback(
         (item) =>
-            setSelected3DFilePivotItem(
-                item.props.itemKey as SelectionModeOf3DFile
-            ),
+            dispatch({
+                type: SET_SELECTED_3D_FILE_PIVOT_ITEM,
+                payload: item.props.itemKey as SelectionModeOf3DFile
+            }),
         []
     );
 
     const resetState = useCallback(() => {
-        setNewSceneName('');
-        setNewSceneDescription('');
-        setNewLatitudeValue(undefined);
-        setNewLatitudeValue(undefined);
+        dispatch({
+            type: RESET_SCENE
+        });
         setIsShowOnGlobeEnabled(isShowOnGlobeEnabled);
-        setNewSceneBlobUrl('');
-        setIsSelectedFileExistInBlob(false);
-        setIsOverwriteFile(false);
-        setBlobsInContainer([]);
-        setSelectedFile(null);
-        setSelected3DFilePivotItem(SelectionModeOf3DFile.FromContainer);
-    }, [isShowOnGlobeEnabled]);
+    }, []);
 
     const isSubmitButtonDisabled = useMemo(() => {
```

Comment:
can we have hte reducer compute the "isValid" state? That can we updated anytime a property is updated and stored on the state there. The reducer alreayd knows all the values so it would make sense to have this computed there.

Reply:
im not clear on how the reducer would continuously calculate the isValid state, check my comment above on the naive way i thought about it

Reply:
it will only need to calculate when particular actions are triggered to update those specific fields it cares about.